src/main/java/uk/gov/ida/verifyserviceprovider/configuration/VerifyServiceProviderConfiguration.java

```diff
@@ -23,7 +24,7 @@
     private PrivateKey samlSigningKey;
     private PrivateKey samlPrimaryEncryptionKey;
     private PrivateKey samlSecondaryEncryptionKey;
-    private MsaMetadataConfiguration msaMetadata;
+    private Optional<MsaMetadataConfiguration> msaMetadata;
```

Comment:
I think I recall hearing Andy say we shouldn't use Optionals in Jackson-serialised classes, as it leads to having multiple ways to represent the absence of a value, which can get messy in Jackson-land.  Nulls are horrible but we're stuck with them in this context (I believe there are workarounds, but I don't know whether we're keen on using them).

Andy, please verify I'm not talking rubbish here - I wish we *could* use Optional!

Reply:
Thanks for your comment as always, I think in this case it may be fine because Jackson is used to deserialisation from verify-service-provider.yml to it's java equivalent VerifyServiceProviderConfiguration which in turn seems to be exclusively used by our code. So there are not external consumers who may assume an absence of data to be equivalent to NULL. So resource endpoints which have to return some data in JSON to be consumed by some client which may not necessarily be Java based would fall into this category. 

But I agree that there may be a case for being consistent. Depending on feedback from Andy, I would be happy to explore your workaround.